scalyr_agent/copying_manager/copying_manager.py

```diff
@@ -99,12 +99,16 @@ def __init__(self, config, api_key_config):
                 # Important for the understanding. When the shared_object_manager is started, it creates a new process.
                 # We use this process as a process for the worker.
 
+                # set the parent process pid for the SharedObjectManager's watchdog
+                # which should detect if parent was killed.
+                SharedObjectManager.parent_pid = os.getpid()
```

Comment:
It seems odd to me from a code perspective to set this as a class instance variable.  I'm going to make some suggestions for refactoring this watch dog as a separate abstraction.  As part of that, I'm going to suggest removing this and instead invoking another method during the worker initialization.